File: 234-palindrome-linked-list/234-palindrome-linked-list.cpp

class Solution {
public:
    bool isPalindrome(ListNode* head) {
        stack<int> s;
        ListNode *slow=head;
        while(slow!=NULL)
        {
                s.push(slow->val);
                slow=slow->next;
            
        }
        while(head!=NULL)
        {int i=s.top();
          s.pop();
          if(head->val!=i)
              return false;
         head=head->next;
            
        }
        return true;
    }
};

File: 234-palindrome-linked-list/234-palindrome-linked-list.java

class Solution {
    public boolean isPalindrome(ListNode head) {
         Stack<Integer> s=new Stack<>();
        ListNode slow=head;
        while(slow!=null)
        {
                s.push(slow.val);
                slow=slow.next;
            
        }
        while(head!=null)
        {int i=s.pop();
          
          if(head.val!=i)
              return false;
         head=head.next;
            
        }
        return true;
    }
}